### PR TITLE
ESP32-S2: Log all Wi-Fi events & remove unneeded call

### PR DESCRIPTION
**Pull request by anecdata**
### __init.c__:

Add logging to the event handler default event to capture any other wi-fi events that occur, to see what else happens in the field that we may want to handle better.

### Radio.c:

This call in `start_station()`:
```
esp_wifi_set_config(WIFI_MODE_STA, &self->sta_config);
```
has the wrong parameter [type] (should be `ESP_IF_WIFI_STA` == 0; `WIFI_MODE_STA` == 1 and sets the config for AP interface type ), and is also not needed here since it's called in `connect()` anyway (and is not needed for scanning or ping).

Alternatively, we could correct the call in `start_station()`, and remove it from `connect()` (`connect()` calls `start_station()`)

All depends on the design intent... how granular we want these functions to be as features like AP, AP+STA, etc. get added. It might be nice to keep clean code demarcation between concepts of netif [de]init, wifi [de]init, wi-fi mode, wifi start/stop, and station [dis]connect.

### CODE REVIEW

File: ports/esp32s2/common-hal/wifi/__init__.c [hunk 1]
```diff
@@ -78,12 +78,15 @@ static void event_handler(void* arg, esp_event_base_t event_base,
...
                 radio->last_disconnect_reason = reason;
                 xEventGroupSetBits(radio->event_group_handle, WIFI_DISCONNECTED_BIT);
+                break;
             }
 
             // Cases to handle later.
             // case WIFI_EVENT_STA_AUTHMODE_CHANGE:
-            default:
+            default: {
+                ESP_EARLY_LOGW(TAG, "event %d 0x%02x", event_id, event_id);
```
Comment (tannewt) [inline ports/esp32s2/common-hal/wifi/__init__.c:87]: We should switch these to non-early logs. That way it won't interrupt another log line. Regular ones are queued I believe.

```suggestion
                ESP_LOGW(TAG, "event %d 0x%02x", event_id, event_id);
```

**Review by tannewt — CHANGES_REQUESTED** [review]
Not early log please.

File: ports/esp32s2/common-hal/wifi/__init__.c (hunk 1 above)
Comment (anecdata) [reply ports/esp32s2/common-hal/wifi/__init__.c]: I changed all of them in the event_handler (I assume that's what you intended), there are none are elsewhere in the file

**Review by tannewt — APPROVED** [review]
Looks good! Thank you!

**Comment by anecdata** [comment]
Partially addresses the logging aspect of [https://github.com/adafruit/circuitpython/issues/3837](https://github.com/adafruit/circuitpython/issues/3837).

**Pull request merged**